inspector/temp.go

```diff
@@ -0,0 +1,203 @@
+package inspector
+
+import (
+	"encoding/json"
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/bisohns/saido/driver"
+	log "github.com/sirupsen/logrus"
+)
+
+type TempMetrics struct {
+	CPUTemp float64
+}
+
+type TempLinux struct {
+	Driver            *driver.Driver
+	CPUTempFilePath   string
+	RawTempDegree     string
+	DisplayTempDegree string
+	Note              string
+	Values            *TempMetrics
+}
+
+type TempDarwin struct {
+	Driver            *driver.Driver
+	CPUTempCommand    string
+	RawTempDegree     string
+	DisplayTempDegree string
+	Note              string
+	Values            *TempMetrics
+}
+
+type TempWin struct {
+	Driver            *driver.Driver
+	CPUTempCommand    string
+	RawTempDegree     string
+	DisplayTempDegree string
+	Note              string
+	Values            *TempMetrics
+}
+
+func (i *TempWin) SetDriver(driver *driver.Driver) {
+	details := (*driver).GetDetails()
+	if !details.IsWindows {
+		panic("Cannot use TempWin outside (windows)")
+	}
+	i.Driver = driver
+}
+
+func (i TempWin) driverExec() driver.Command {
+	return (*i.Driver).RunCommand
+}
+
+func (i *TempWin) Parse(output string) {
+	// var values []TempMetrics
+	log.Debug("Parsing output string in TempWin inspector")
+	lineChar := "\r"
+	output = strings.TrimPrefix(output, lineChar)
+	output = strings.TrimSuffix(output, lineChar)
+	lines := strings.Split(output, "\n")
+	if len(lines) < 1 {
+		log.Fatalf(`Error Parsing Temperature `)
+	}
+	farenValue, err := strconv.ParseFloat(lines[1], 64)
+	if err != nil {
+		log.Fatalf(`Error Parsing Temperature: %s `, err)
+	}
+	celsValue := (farenValue * 10.0) - 27315.0
+	i.Values = &TempMetrics{
+		CPUTemp: celsValue,
+	}
+}
+
+func (i *TempWin) Execute() ([]byte, error) {
+	output, err := i.driverExec()(i.CPUTempCommand)
+	if err == nil {
+		i.Parse(output)
+		return json.Marshal(i.Values)
+	}
+	return []byte(""), err
+}
+
+func (i *TempLinux) SetDriver(driver *driver.Driver) {
+	details := (*driver).GetDetails()
+	if !details.IsLinux {
+		panic("Cannot use TempLinux outside (linux)")
+	}
+	i.Driver = driver
+}
+
+func (i TempLinux) driverExec() driver.Command {
+	return (*i.Driver).RunCommand
+}
+
+func (i *TempLinux) Parse(output string) {
+	log.Debug("Parsing output string in Temp inspector")
+	output = strings.ReplaceAll(output, " ", "")
+	value, err := strconv.ParseFloat(output, 64)
+	if err != nil {
+		log.Fatalf(`Error Parsing Temperature: %s `, err)
```

Comment:
Let's try not to fatally kill everybody.... just turn this to an Errorf on parsing issues

Reply:
Same for other Fatals ... whenever the function can't return an error, just use Errorf but if it can return an error then return it